pkg/apis/openebs.io/scp/v1alpha1/capabilties.go

```diff
@@ -0,0 +1,125 @@
+type Capabilities struct {
+	// DataStorage capabilities
+	DataStorage *DataStorageCapabilities `json:"dataStorage"`
+
+	// Data security capabilities
+	// +nullable
+	DataSecurity *DataSecurityCapabilities `json:"dataSecurity,omitempty"`
+
+	// IO connectivity capabilities
+	// +nullable
+	IOConnectivity *IOConnectivityCapabilities `json:"IOConnectivity,omitempty"`
+
+	// IO performance capabilities
+	IOPerformance *IOPerformanceCapabilities `json:"IOPerformance"`
+
+	// DataProtection capabilities
+	// +nullable
+	DataProtection *DataProtectionCapabilities `json:"dataProtection,omitempty"`
+}
+
+// DataStorageCapabilities defines data storage capabilities for a pool or volume
+type DataStorageCapabilities struct {
+	//AccessModes contains the actual access modes
+	AccessModes []AccessMode `json:"accessModes"`
+
+	// ProvisioningPolicy defines provisioning policy type
+	ProvisioningPolicy []string `json:"provisioningPolicy"`
+
+	// MultiPathing to be supported or not.
+	// +nullable
+	MultiPathing []string `json:"multiPathing,omitempty"`
+
+	// Compression to be supported or not, if yes the algorithm
+	//	+nullable
+	Compression []string `json:"compression,omitempty"`
+
+	// Deduplication to be supported or not.
+	// +nullable
+	Deduplication []string `json:"deduplication,omitempty"`
+}
+
+// AccessMode of the pool or volume
+type AccessMode string
+
+const (
+	// ReadWriteOnce can be mounted in read/write mode to exactly 1 host
+	ReadWriteOnce AccessMode = "ReadWriteOnce"
+
+	// ReadOnlyMany can be mounted in read-only mode to many hosts
+	ReadOnlyMany AccessMode = "ReadOnlyMany"
+
+	// ReadWriteMany can be mounted in read/write mode to many hosts
+	ReadWriteMany AccessMode = "ReadWriteMany"
+)
+
+
+// DataSecurityCapabilities defines media encryption algorithms and data sanitization policy.
+type DataSecurityCapabilities struct {
+
+	// MediaEncryption to be supported or not, if yes the algorithms
+	// +nullable
+	MediaEncryption []string `json:"mediaEncryption,omitempty"`
+
+	// DataSanitizationPolicy to be supported, viz Clear, CryptographicErase
+	// +nullable
+	DataSanitizationPolicy []string `json:"dataSanitizationPolicy,omitempty"`
+}
+
+// IOConnectivityCapabilities defines access protocols to be supported by the pool or volume
+type IOConnectivityCapabilities struct {
+	// AccessProtocols to be supported, viz NVMe, NVMeOverFabrics
+	// +nullable
+	AccessProtocols []string `json:"accessProtocols,omitempty"`
+}
+
+// IOPerformanceCapabilities defines IO performance capabilities for the pool or volume.
+type IOPerformanceCapabilities struct {
+	// AverageIOOperationLatencyMicroseconds to be supported or not.
+	// +nullable
+	AverageIOOperationLatencyMicroseconds *uint64 `json:"averageIOOperationLatencyMicroseconds,omitempty"`
+
+	// maxIOOperationsPerSecondPerTerabyte to be supported or not.
+	// +nullable
+	maxIOOperationsPerSecondPerTerabyte *uint64 `json:"maxIOOperationsPerSecondPerTerabyte,omitempty"`
```

Comment:
Can `maxIOOperationsPerSecondPerTerabyte` be changed to `maxIOPSPerTerabyte`?

Reply:
Also why Terabyte why not Gigabyte?

Reply:
Do we want to keep this field as unexported?

Reply:
> Also why Terabyte why not Gigabyte?

Not sure, created it as per the Design Document.

Reply:
> Can `maxIOOperationsPerSecondPerTerabyte` be changed to `maxIOPSPerTerabyte`?

Can we rename it to `maxIOPSPerGB`?

Reply:
Usually, persistent drives come in TBs nowadays and maxIOPSPerTB seems more apt.

Reply:
>Do we want to keep this field as unexported?

All the fields should be exported.

Reply:
> All the fields should be exported
It starts with a lowercase, that's why mentioned the same. It should be upper case in that case.